src/extractor/ts/src/cloesce.ts

```diff
@@ -153,45 +149,58 @@ export async function cloesce(
     }
 
     instance = successfulModel.value;
-  } else {
-    instance = constructorRegistry[modelMeta.name];
   }
 
-  // 5. Method Dispatch
+  // Dispatch a method on the model and return the result
   return toResponse(
     await methodDispatch(instance, methodMeta, requestParamMap, d1),
   );
 }
 
 // TODO: In the previous version, we would walk a generated trie
 // This is more hardcode-y, and I'm not sure it will hold up to time.
+/**
+ * Matches a request to a method on a model.
+ * @param api_route The route from the domain to the actual API, ie https://foo.com/route/to/api => route/to/api/
+ * @returns 404 or a `MatchedRoute`
+ */
 function matchRoute(
   request: Request,
   api_route: string,
   cidl: MetaCidl,
-): Either<HttpResult, Route> {
+): Either<HttpResult, MatchedRoute> {
   const url = new URL(request.url);
 
-  const err = left(error_state(404, `Path not found ${url.pathname}`));
-
-  if (!url.pathname.startsWith(api_route)) return err;
+  const notFound = (e: string) =>
+    left(error_state(404, `Path not found: ${e} ${url.pathname}`));
 
   const routeParts = url.pathname
     .slice(api_route.length)
     .split("/")
     .filter(Boolean);
 
-  if (routeParts.length < 2) return err;
+  if (routeParts.length < 2) {
+    return notFound("Expected /model/method or /model/:id/method");
+  }
 
+  // Attempt to extract from routeParts
   const modelName = routeParts[0];
   const methodName = routeParts[routeParts.length - 1];
   const id = routeParts.length === 3 ? routeParts[1] : null;
 
   const modelMeta = cidl.models[modelName];
-  if (!modelMeta) return err;
+  if (!modelMeta) {
+    return notFound(`Unknown model ${modelName}`);
+  }
 
   const methodMeta = modelMeta.methods[methodName];
-  if (!methodMeta) return err;
+  if (!methodMeta) {
+    return notFound(`Unknown method ${modelName}.${methodName}`);
+  }
+
+  if (request.method !== methodMeta.http_verb) {
+    return notFound("Unmatched HTTP method");
```

Comment:
Something worth thinking about @KyanPepper @nokuyo , do these messages pose a security risk? Before I just had a generic "Invalid Request" or "Not Found", but for the tests I needed more granularity to make sure we are hitting the right escape states.

Someone could use these messages to effectively "port scan" for models and their methods. When we implement authentication, we might want to revisit this and make sure we don't leak authenticated models, methods and their params.